doc/design/show_train_stmt.md

```diff
@@ -22,31 +21,54 @@ mysql> SHOW CREATE TABLE `test`;
 
 ## Grammar
 ```
-SHOW TRAIN table_where_model_stored;
+SHOW TRAIN table_where_model_stored ;
 ```
 where
-- `table_where_model_stored` must be a table which stores the trained model in MySQL or Hive
+- `table_where_model_stored` must be a table which stores the trained model in MySQL or Hive. It should be a qualified table name like `{db_name}.{table_name}` if you are not in a DB selected context.
 
 This query will show the original train sql statement like:
 ```
 +-------+------------------------------------------------------------------+
-| Table | Train Statement
+| TABLE | TRAIN STATEMENT
 +-------+------------------------------------------------------------------+
 | test  | SELECT * FROM train to TRAIN xgboost.gbtree with learning_rate=0.4,
  objective=multi:softmax, num_class=3 LABEL class INTO test;
-+-------+-----------------------------------------------------------------+
++-------+------------------------------------------------------------------+
 ```
-In the future we may support to show more metadata other than the original sql statement.
+In the future, we may support to show more metadata other than the original training sql statement.
 
 ## Implementation
-
-- Extend the SQLFlow parser with our `SHOW TRAIN` statement
-- Branch to execute a `ShowTrain` cmd in executor, this cmd will first read the model on a `sqlfs` and  then extract the original train statement
+- Extend the SQLFlow parser with our `SHOW TRAIN` statement. First, we need to add a key word `SHOW` to our extended grammar. In addition, `SHOW TRAIN` is not like our train/predict/explain statements which all share a format `SELECT ... TO ...` in which there is a **standard** `SELECT` part at the front and an **extended** `TO ...` part at the end. With this definition, our extending statement has no standard part. So, we have to modify the parse process slightly. The pseudo code is like below:
+```go
+func Parse(program string) ([]*SQLFlowStmt, error) {
+  all := make([]*SQLFlowStmt)
+  for program != "" {
+    standard, err := tryExtractStandardPart(program)
+    if err == nil {
+      all = append(all, standard)
+    } else if extended, e := tryParseByExtendedGrammer(program); e == nil {
+      // Maybe it is a pure extended stmt with no standard part (say, `SHOW TRAIN...`), so try to use the extended parser
+      // In addition, the program should not contains a 'TO...' only, although our extended parser accepts this form
+      if isValidPureExtended(extended) {
+        all = append(all, extended)
+      } else {
+        return nil, err
+      }
+    } else {
+      return nil, err
+    }
+    program = eatConsumedPrefix(program)
+  }
+  return all, nil
+}
+```
+- Branch to execute a `ShowTrain` cmd in executor, this cmd will first read the model on a `sqlfs` and then extract the original train statement. We can not get the train statement from models saved on OSS because they have not been saved with the models yet. This can be implemented next time.
```

Comment:
Please add an issue for putting the train statements into the OSS and update this document after fix the issue.

Reply:
Done, take a look at #2085